pennylane/ops/op_math/composite.py

```diff
@@ -112,13 +112,18 @@ def _op_symbol(self) -> str:
     @property
     def data(self):
         """Create data property"""
-        return [op.data for op in self]
+        data = []
+        for op in self:
+            data.extend(op.data)
+        return data
 
     @data.setter
     def data(self, new_data):
         """Set the data property"""
-        for new_entry, op in zip(new_data, self):
-            op.data = new_entry
+        for op in self:
+            if op.num_params > 0:
+                op.data = new_data[: op.num_params]
+                new_data = new_data[op.num_params :]
```

Comment:
```suggestion
    op_num_params = op.num_params
    if op_num_params > 0:
        op.data = new_data[: op_num_params]
        new_data = new_data[op_num_params :]
```
This seems to be a bit faster, since it only needs to call `op.num_params` once.

Reply:
I took this, just made another comment with correct whitespace